src/firestore/indexes.ts

```diff
@@ -0,0 +1,560 @@
+import * as clc from "cli-color";
+
+import * as api from "../api";
+import * as FirebaseError from "../error";
+import * as logger from "../logger";
+import * as utils from "../utils";
+import * as validator from "./validator";
+
+import * as API from "./indexes-api";
+import * as Spec from "./indexes-spec";
+
+// projects/$PROJECT_ID/databases/(default)/collectionGroups/$COLLECTION_GROUP_ID/indexes/$INDEX_ID
+const INDEX_NAME_REGEX = /projects\/([^\/]+?)\/databases\/\(default\)\/collectionGroups\/([^\/]+?)\/indexes\/([^\/]*)/;
+
+// projects/$PROJECT_ID/databases/(default)/collectionGroups/$COLLECTION_GROUP_ID/fields/$FIELD_ID
+const FIELD_NAME_REGEX = /projects\/([^\/]+?)\/databases\/\(default\)\/collectionGroups\/([^\/]+?)\/fields\/([^\/]*)/;
+
+interface IndexName {
+  projectId: string;
+  collectionGroupId: string;
+  indexId: string;
+}
+
+interface FieldName {
+  projectId: string;
+  collectionGroupId: string;
+  fieldPath: string;
+}
+
+export class FirestoreIndexes {
+  /**
+   * Deploy an index specification to the specified project.
+   * @param project the Firebase project ID.
+   * @param indexes an array of objects, each will be validated and then converted
+   * to an {@link IndexSpecEntry}.
+   */
+  async deploy(project: string, indexes: any[], fieldOverrides: any[]): Promise<any> {
```

Comment:
does this really return `any`? looks to me like it returns `void` (aka `undefined`)